cpu/msp430fxyz/periph/gpio.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_msp430fxyz
+ * @{
+ *
+ * @file
+ * @brief       Low-level GPIO driver implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "bitarithm.h"
+#include "periph/gpio.h"
+
+/**
+ * @brief   Number of possible interrupt lines: 2 ports * 8 pins
+ */
+#define ISR_NUMOF           (16U)
+
+/**
+ * @brief   Number of pins on each port
+ */
+#define PINS_PER_PORT       (8U)
+
+/**
+ * @brief   Datatype to use for saving the interrupt contexts
+ */
+typedef struct {
+    gpio_cb_t cb;       /**< callback to call on GPIO interrupt */
+    void *arg;          /**< argument passed to the callback */
+} isr_ctx_t;
+
+/**
+ * @brief   Interrupt context for each interrupt line
+ */
+static isr_ctx_t isr_ctx[ISR_NUMOF];
+
+
+static msp_port_t *_port(gpio_t pin)
+{
+    switch (pin >> 8) {
+        case 1:
+            return PORT_1;
```

Comment:
What is https://github.com/RIOT-OS/RIOT/pull/3760/files#diff-bedb5324b18a3d4064038b8dcc94f1f9R75 then?